code/modules/mining/lavaland/megafauna_loot.dm

```diff
@@ -1115,3 +1115,168 @@
+	return FALSE
+
+/obj/item/organ/brain/cybernetic/ai/proc/owner_clicked(datum/source, atom/location, control, params, mob/user)
+	SIGNAL_HANDLER
+	if(!isAI(user))
+		return
+	var/list/lines = list()
+	lines += span_bold("[owner]")
+	lines += "Target is currently [!HAS_TRAIT(owner, TRAIT_INCAPACITATED) ? "functional" : "incapacitated"]"
+	lines += "Estimated organic/inorganic integrity: [owner.health]"
+	if(is_sufficiently_augmented())
+		lines += "<a href='byond://?src=[REF(src)];ai_take_control=[REF(user)]'>[span_boldnotice("Take control?")]</a><br>"
+	else
+		lines += span_warning("Organic organs detected. Robotic organs only, cannot take over.")
+
+	to_chat(user, boxed_message(jointext(lines, "\n")), type = MESSAGE_TYPE_INFO)
+
+/obj/item/organ/brain/cybernetic/ai/Topic(href, href_list)
+	..()
+	if(!href_list["ai_take_control"] || !is_sufficiently_augmented())
+		return
+	var/mob/living/silicon/ai/AI = locate(href_list["ai_take_control"]) in GLOB.silicon_mobs
+	if(isnull(AI))
+		return
+	if(AI.controlled_equipment)
+		to_chat(AI, span_warning("You are already loaded into an onboard computer!"))
+		return
+	if(!GLOB.cameranet.checkCameraVis(owner))
+		to_chat(AI, span_warning("Target is no longer near active cameras."))
+		return
+	if(!isturf(AI.loc))
+		to_chat(AI, span_warning("You aren't in your core!"))
+		return
+
+	RegisterSignal(owner, COMSIG_LIVING_DEATH, PROC_REF(undeploy))
+	AI.deployed_shell = owner
+	deploy_init(AI)
+	ADD_TRAIT(AI.mind, TRAIT_UNCONVERTABLE, ORGAN_TRAIT)
+	AI.mind.transfer_to(owner)
+
+/obj/item/organ/brain/cybernetic/ai/proc/deploy_init(mob/living/silicon/ai/AI)
+	//todo camera maybe
+	mainframe = AI
+	RegisterSignal(AI, COMSIG_QDELETING, PROC_REF(ai_deleted))
+	undeployment_action.Grant(owner)
+	update_med_hud_status(owner)
+	to_chat(owner, span_boldbig("You are still considered a silicon/cyborg/AI. Follow your laws."))
+
+/obj/item/organ/brain/cybernetic/ai/proc/undeploy(datum/source)
+	SIGNAL_HANDLER
+	if(!owner?.mind || !mainframe)
+		return
+	UnregisterSignal(src, list(COMSIG_LIVING_DEATH, COMSIG_QDELETING))
+	mainframe.redeploy_action.Remove(mainframe)
+	mainframe.redeploy_action.last_used_shell = null
+	owner.mind.transfer_to(mainframe)
+	mainframe.deployed_shell = null
+	undeployment_action.Remove(owner)
+	if(mainframe.laws)
+		mainframe.laws.show_laws(mainframe)
+	if(mainframe.eyeobj)
+		mainframe.eyeobj.setLoc(loc)
+	REMOVE_TRAIT(mainframe.mind, TRAIT_UNCONVERTABLE, ORGAN_TRAIT)
+	mainframe = null
+	update_med_hud_status(owner)
+
+/obj/item/organ/brain/cybernetic/ai/proc/is_sufficiently_augmented()
+	var/mob/living/carbon/carb_owner = owner
+	. = TRUE
+	if(!istype(carb_owner))
+		return
+	for(var/obj/item/organ/organ as anything in carb_owner.organs)
+		if(!IS_ROBOTIC_ORGAN(organ) && !istype(organ, /obj/item/organ/tongue)) //tongues are not in the exosuit fab and nobody is going to bother to find them so
```

Comment:
In my opinion, this should also exclude external organs, which shouldn't really be in the way of anything that can be printed by the exofab.